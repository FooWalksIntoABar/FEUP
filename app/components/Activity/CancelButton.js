import React from 'react'
import { Image, StyleSheet, TouchableOpacity, View } from 'react-native'
import PropTypes from 'prop-types'
import { buttonStyle } from '../../styles/Activity.style'
import { red } from '../../styles/Colors'
import Images from '../../assets/images/images'
import Layout from '../../constants/Layout'

import * as Progress from 'react-native-progress'

export class CancelButton extends React.Component {
  constructor (props) {
    super(props)
    this.handleButtonPress = this.handleButtonPress.bind(this)
    this.handleButtonRelease = this.handleButtonRelease.bind(this)

    this.state = {
      progress: 0,
      isCancelling: false
    }
  }

  handleButtonPress () {
    this.timer = setInterval(() => {
      if (this.state.progress >= 0.99) {
        this.props.cancelActivity()
      }

      this.setState(() => {
        return {
          progress: this.state.progress + 0.01,
          isCancelling: true
        }
      })
    }, 10)
  }

  handleButtonRelease () {
    clearInterval(this.timer)
    this.setState(() => {
      return {
        progress: 0,
        isCancelling: false
      }
    })
  }

  componentWillUnmount () {
    clearInterval(this.timer)
  }

  render () {
    return (
      <View>
        { this.state.isCancelling &&
          <Progress.Bar
            width={Layout.window.height * 0.12 / 1.5}
            progress={this.state.progress}
          />
        }
        <TouchableOpacity
          activeOpacity={0.8}
          style={[styles.button, this.props.style]}
          onPressIn={this.handleButtonPress}
          onPressOut={this.handleButtonRelease} >
          <Image
            style={styles.image}
            resizeMode={'contain'}
            source={Images.ui.cancel} />
        </TouchableOpacity>
      </View>
    )
  }
}

CancelButton.propTypes = {
  cancelActivity: PropTypes.func.isRequired,
  style: PropTypes.number.isRequired
}

const styles = StyleSheet.create({
  button: {
    ...buttonStyle,
    backgroundColor: red
  },
  image: {
    height: '35%',
    width: '35%',
    tintColor: '#fff'
  }
})
