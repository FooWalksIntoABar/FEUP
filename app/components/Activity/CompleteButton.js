import React from 'react'
import { Image, StyleSheet, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import activityStyles, { buttonStyle } from '../../styles/Activity.style'
import { green, gray } from '../../styles/Colors'
import Images from '../../assets/images/images'

export class CompleteButton extends React.Component {
  render () {
    return (
      <TouchableOpacity
        activeOpacity={0.8}
        style={[(this.props.isCompletable ? styles.buttonActive : styles.buttonDisabled), activityStyles.largeButton]}
        onPress={this.props.completeActivity}
        disabled={!this.props.isCompletable} >
        <Image
          style={styles.image}
          resizeMode={'contain'}
          source={Images.ui.confirm} />
      </TouchableOpacity>
    )
  }
}

CompleteButton.propTypes = {
  completeActivity: PropTypes.func.isRequired,
  isCompletable: PropTypes.bool.isRequired
}

const styles = StyleSheet.create({
  buttonActive: {
    ...buttonStyle,
    backgroundColor: green
  },
  buttonDisabled: {
    ...buttonStyle,
    backgroundColor: gray
  },
  image: {
    height: '50%',
    width: '50%',
    tintColor: '#fff'
  }
})
