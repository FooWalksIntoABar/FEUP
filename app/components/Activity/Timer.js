import React from 'react'
import { Text } from 'react-native'
import PropTypes from 'prop-types'

export class Timer extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      minutes: '0',
      seconds: '00'
    }
  }

  componentWillReceiveProps (props) {
    let minutes = Math.floor(props.elapsedTime / 60)

    let seconds = Math.floor(props.elapsedTime % 60)
    if (seconds < 10) seconds = '0' + seconds

    this.setState(() => ({ minutes: minutes, seconds: seconds }))
  }

  render () {
    return (
      <Text style={this.props.style}>{`${this.state.minutes}:${this.state.seconds}`}</Text>
    )
  }
}

Timer.propTypes = {
  elapsedTime: PropTypes.number.isRequired,
  style: PropTypes.object.isRequired
}
