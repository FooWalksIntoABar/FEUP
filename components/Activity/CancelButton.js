import React from 'react'
import { Image, StyleSheet, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { buttonStyle } from '../../styles/Activity.style'
import { red } from '../../styles/Colors'

export class CancelButton extends React.Component {
  render () {
    return (
      <TouchableOpacity
        style={[styles.button, this.props.style]}
        onPress={this.props.cancelActivity} >
        <Image
          style={styles.image}
          resizeMode={'contain'}
          source={require('../../assets/images/nav-cancel.png')} />
      </TouchableOpacity>
    )
  }
}

CancelButton.propTypes = {
  cancelActivity: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
  button: {
    ...buttonStyle,
    backgroundColor: red
  },
  image: {
    height: '35%',
    width: '35%',
    tintColor: '#fff'
  }
})
