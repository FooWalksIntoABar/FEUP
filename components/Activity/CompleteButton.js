import React from 'react'
import { Image, StyleSheet, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { buttonStyle } from '../../styles/Activity.style'
import { green, gray } from '../../styles/Colors'
import Images from '../../assets/images/images'

export class CompleteButton extends React.Component {
  render () {
    return (
      <TouchableOpacity
        activeOpacity={0.8}
        style={[(this.props.isCompletable ? styles.buttonActive : styles.buttonDisabled), this.props.style]}
        onPress={this.props.completeActivity}
        disabled={!this.props.isCompletable} >
        <Image
          style={styles.image}
          resizeMode={'contain'}
          source={Images.ui.confirm} />
      </TouchableOpacity>
    )
  }
}

CompleteButton.propTypes = {
  completeActivity: PropTypes.func.isRequired,
  isCompletable: PropTypes.bool
}

const styles = StyleSheet.create({
  buttonActive: {
    ...buttonStyle,
    backgroundColor: green
  },
  buttonDisabled: {
    ...buttonStyle,
    backgroundColor: gray
  },
  image: {
    height: '50%',
    width: '50%',
    tintColor: '#fff'
  }
})
