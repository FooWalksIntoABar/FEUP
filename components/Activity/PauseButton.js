import React from 'react'
import { Image, StyleSheet, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { buttonStyle } from '../../styles/Activity.style'

export class PauseButton extends React.Component {
  render () {
    return (
      <TouchableOpacity
        style={[styles.button, this.props.style]}
        onPress={this.props.pauseActivity} >
        <Image
          style={styles.image}
          resizeMode={'contain'}
          source={require('../../assets/images/nav-pause.png')} />
      </TouchableOpacity>
    )
  }
}

PauseButton.propTypes = {
  // resumeActivity: PropTypes.func.isRequired,
  pauseActivity: PropTypes.func.isRequired,
  // isPaused: PropTypes.bool
}

const styles = StyleSheet.create({
  button: {
    ...buttonStyle,
    backgroundColor: '#dba159'
  },
  image: {
    height: '50%',
    width: '50%',
    tintColor: '#fff'
  }
})
