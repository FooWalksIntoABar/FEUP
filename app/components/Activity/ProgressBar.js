import React from 'react'
import { View, Vibration, Animated } from 'react-native'
import { Audio } from 'expo'
import PropTypes from 'prop-types'
import styles, { buttonHeight, timerBar } from '../../styles/Activity.style'
import { Timer } from './Timer'
import * as colors from '../../styles/Colors'
import Layout from '../../constants/Layout'

import * as Progress from 'react-native-progress'

const SUCCESS_COLOR = 'red'
const ORIGINAL_COLOR = 'transparent'
const ORIGINAL_VALUE = 0
const SUCCESS_VALUE = 1

export class ProgressBar extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      color: colors.darkGray,
      playedFeedback: false,
      feedbackCycles: 0
    }
    this.times = props.activityTimes
    this.frequency = 0
    this.feedbackBorderWidth = 0
    this.feedbackElevation = new Animated.Value(6)
    this.borderColor = new Animated.Value(ORIGINAL_VALUE)
    this.borderWidth = new Animated.Value(0)
  }

  componentWillUnmount () {
    Vibration.cancel()
    clearInterval(this.interval)
  }

  componentDidMount () {
    this.assignFeedbackFrequency()
  }

  assignFeedbackFrequency () {
    switch (this.props.feedbackFrequency) {
      case 'slow':
        this.frequency = 3
        break
      case 'normal':
        this.frequency = 2
        break
      case 'fast':
        this.frequency = 1
        break
      default:
        break
    }
  }

  playSound = async () => {
    const soundObject = new Audio.Sound()
    try {
      await soundObject.loadAsync(require('../../assets/sounds/alarmclockbeeps.mp3'))
      this.audioPlayer1 = soundObject
      this.audioPlayer1.playAsync()
      this.audioPlayer1.setPositionAsync(0)
      this.audioPlayer1.setRateAsync(3, false)
    } catch (error) { }
  }

  borderFeedback = () => {
    Animated.sequence([
      Animated.parallel([
        Animated.timing(this.feedbackElevation, {
          duration: 700,
          toValue: 20
        }),
        Animated.timing(this.borderColor, {
          duration: 700,
          toValue: SUCCESS_VALUE
        }),
        Animated.timing(this.borderWidth, {
          duration: 700,
          toValue: 3
        })
      ]),

      Animated.parallel([
        Animated.timing(this.feedbackElevation, {
          duration: 700,
          toValue: 6
        }),
        Animated.timing(this.borderColor, {
          duration: 700,
          toValue: ORIGINAL_VALUE
        }),
        Animated.timing(this.borderWidth, {
          duration: 700,
          toValue: 0
        })
      ])
    ]).start()
  }

  activityFeedback () {
    if (this.props.activityFeedback === 'visual') {
      this.borderFeedback()
    } else if (this.props.activityFeedback === 'sound') {
      this.playSound()
    } else if (this.props.activityFeedback === 'vibration') {
      try {
        Vibration.vibrate(250, true)
      } catch (error) {
        console.log('This device cannot vibrate')
      }
    }
  }

  componentWillReceiveProps (props) {
    console.log('this.state.playedFeedback = ' + this.state.playedFeedback)
    if (props.elapsedTime >= this.times.goal + (this.times.max - this.times.goal) / 2) {
      this.setState(() => ({ color: colors.red }))

      if (!this.state.playedFeedback || this.state.feedbackCycles % (15 * this.frequency) === 0) {
        this.activityFeedback()
        this.setState(() => ({ playedFeedback: true, feedbackCycles: this.state.feedbackCycles + 1 }))
      } else {
        this.setState(() => ({ feedbackCycles: this.state.feedbackCycles + 1 }))
      }
    } else if (props.elapsedTime >= this.times.goal) this.setState(() => ({ color: colors.yellow }))
    else if (props.elapsedTime > this.times.min) this.setState(() => ({ color: colors.green }))

    if (props.isPaused) {
      this.setState(() => ({ color: colors.darkGray }))
      Vibration.cancel()
    }
  }

  render () {
    let feedbackBorderColor = this.borderColor.interpolate({
      inputRange: [ORIGINAL_VALUE, SUCCESS_VALUE],
      outputRange: [ORIGINAL_COLOR, SUCCESS_COLOR]
    })

    return (
      <Animated.View style={[styles.progressBarContainer, { elevation: this.feedbackElevation, borderColor: feedbackBorderColor, borderWidth: this.borderWidth }]}>
        <Progress.Bar
          width={null}
          height={buttonHeight / 1.5}
          borderRadius={Layout.window.height * 0.24 / 3}
          borderWidth={this.feedbackBorderWidth}
          borderColor={'red'}
          progress={this.props.elapsedTime / this.times.max}
          color={this.state.color}
        />
        <View style={[{ left: `${this.times.min / this.times.max * 100}%` }, styles.progressBarDivider]} />
        <View style={[{ left: `${this.times.goal / this.times.max * 100}%` }, styles.progressBarDivider]} />
        <View style={[{ left: `${(this.times.goal + (this.times.max - this.times.goal) / 2) / this.times.max * 100}%` }, styles.progressBarDivider]} />
        {this.props.showTimer && <Timer style={timerBar} remainingTime={this.times.max - this.props.elapsedTime} />}
      </Animated.View>
    )
  }
}

ProgressBar.propTypes = {
  elapsedTime: PropTypes.number.isRequired,
  activityTimes: PropTypes.object.isRequired,
  isPaused: PropTypes.bool.isRequired,
  showTimer: PropTypes.bool.isRequired,
  activityFeedback: PropTypes.string.isRequired,
  feedbackFrequency: PropTypes.string.isRequired
}
