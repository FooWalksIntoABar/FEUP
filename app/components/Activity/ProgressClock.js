import React from 'react'
import { Vibration, Animated } from 'react-native'
import { Audio } from 'expo'
import PropTypes from 'prop-types'
import styles, { clockHeight, timerClock } from '../../styles/Activity.style'
import { Timer } from './Timer'
import * as colors from '../../styles/Colors'

import * as Progress from 'react-native-progress'

const SUCCESS_COLOR = 'red'
const ORIGINAL_COLOR = 'black'
const ORIGINAL_VALUE = 0
const SUCCESS_VALUE = 1

export class ProgressClock extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      color: colors.darkGray,
      playedFeedback: false,
      feedbackCycles: 0
    }
    this.times = props.activityTimes
    this.feedbackBorderWidth = 0
    this.feedbackElevation = new Animated.Value(6)
    this.borderColor = new Animated.Value(ORIGINAL_VALUE)
    this.borderWidth = new Animated.Value(0)
  }

  componentWillUnmount () {
    Vibration.cancel()
    clearInterval(this.interval)
  }

  playSound = async () => {
    const soundObject = new Audio.Sound()
    try {
      await soundObject.loadAsync(require('../../assets/sounds/alarmclockbeeps.mp3'))
      this.audioPlayer1 = soundObject
      this.audioPlayer1.playAsync()
      this.audioPlayer1.setPositionAsync(0)
      this.audioPlayer1.setRateAsync(3, false)
    } catch (error) { }
  }

  borderFeedback = () => {
    Animated.sequence([
      Animated.parallel([
        Animated.timing(this.feedbackElevation, {
          duration: 700,
          toValue: 20
        }),
        Animated.timing(this.borderColor, {
          duration: 700,
          toValue: SUCCESS_VALUE
        }),
        Animated.timing(this.borderWidth, {
          duration: 700,
          toValue: 3
        })
      ]),

      Animated.parallel([
        Animated.timing(this.feedbackElevation, {
          duration: 700,
          toValue: 6
        }),
        Animated.timing(this.borderColor, {
          duration: 700,
          toValue: ORIGINAL_VALUE
        }),
        Animated.timing(this.borderWidth, {
          duration: 700,
          toValue: 0
        })
      ])
    ]).start()
  }

  activityFeedback () {
    if (this.props.activityFeedback === 'visual') {
      this.borderFeedback()
    } else if (this.props.activityFeedback === 'sound') {
      this.playSound()
    } else if (this.props.activityFeedback === 'vibration') {
      Vibration.vibrate(250, true)
    }
  }

  componentWillReceiveProps (props) {
    if (props.elapsedTime >= this.times.goal + (this.times.max - this.times.goal) / 2) {
      this.setState(() => ({ color: colors.red }))

      if (!this.state.playedFeedback || this.state.feedbackCycles % 25 === 0) {
        this.activityFeedback()
        this.setState(() => ({ playedFeedback: true, feedbackCycles: this.state.feedbackCycles + 1 }))
      } else {
        this.setState(() => ({ feedbackCycles: this.state.feedbackCycles + 1 }))
      }
    } else if (props.elapsedTime >= this.times.goal) this.setState(() => ({ color: colors.yellow }))
    else if (props.elapsedTime > this.times.min) this.setState(() => ({ color: colors.green }))

    if (props.isPaused) this.setState(() => ({ color: colors.darkGray }))
  }

  render () {
    let feedbackBorderColor = this.borderColor.interpolate({
      inputRange: [ORIGINAL_VALUE, SUCCESS_VALUE],
      outputRange: [ORIGINAL_COLOR, SUCCESS_COLOR]
    })

    return (
      <Animated.View style={[styles.progressClockContainer, { elevation: this.feedbackElevation, backgroundColor: feedbackBorderColor }]}>
        <Progress.Circle
          progress={this.props.elapsedTime / this.times.max}
          size={clockHeight}
          thickness={clockHeight / 2}
          color={this.state.color}
          unfilledColor={colors.gray}
          borderWidth={0}
        />
        {this.props.showTimer && <Timer style={timerClock} elapsedTime={this.props.elapsedTime} />}
      </Animated.View>
    )
  }
}

ProgressClock.propTypes = {
  elapsedTime: PropTypes.number.isRequired,
  activityTimes: PropTypes.object.isRequired,
  isPaused: PropTypes.bool.isRequired,
  showTimer: PropTypes.bool.isRequired,
  activityFeedback: PropTypes.string.isRequired
}
