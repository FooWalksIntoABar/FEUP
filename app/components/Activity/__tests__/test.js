import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { activities } from '../../../__tests__/mockTestData'

import { CancelButton } from '../CancelButton'
import { CompleteButton } from '../CompleteButton'
import { PauseButton } from '../PauseButton'
import { ProgressBar } from '../ProgressBar'
import { ProgressClock } from '../ProgressClock'
import { Timer } from '../Timer'

configure({ adapter: new Adapter() })

describe('Activity components', () => {
  it('renders CancelButton correctly', () => {
    const wrapper = shallow(<CancelButton
      cancelActivity={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.instance().handleButtonPress()
    wrapper.instance().handleButtonRelease()
    wrapper.setState({ isCancelling: true })
    wrapper.instance().intervalFunction()
    wrapper.setState({ progress: 1.1 })
    wrapper.instance().intervalFunction()
    wrapper.unmount()
  })

  it('renders CompleteButton correctly', () => {
    const wrapper = shallow(<CompleteButton
      completeActivity={jest.fn()}
      isCompletable />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ isCompletable: false })
    wrapper.unmount()
  })

  it('renders PauseButton correctly', () => {
    const wrapper = shallow(<PauseButton
      resumeActivity={jest.fn()}
      pauseActivity={jest.fn()}
      isPaused />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ isPaused: false })
    wrapper.unmount()
  })

  it('renders ProgressBar correctly', () => {
    const wrapper = shallow(<ProgressBar
      elapsedTime={100}
      activityTimes={activities[0].time}
      isPaused={false}
      showTimer={false}
      activityFeedback={'visual'}
      feedbackFrequency={'normal'} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders ProgressClock correctly', () => {
    const wrapper = shallow(<ProgressClock
      elapsedTime={100}
      activityTimes={activities[0].time}
      isPaused={false}
      showTimer={false}
      activityFeedback={'visual'}
      feedbackFrequency={'normal'} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders Timer correctly', () => {
    const wrapper = shallow(<Timer
      remainingTime={100}
      style={{}} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ remainingTime: 0 })
    wrapper.setProps({ remainingTime: -10 })
    wrapper.setProps({ remainingTime: 13 })
    wrapper.unmount()
  })
})
