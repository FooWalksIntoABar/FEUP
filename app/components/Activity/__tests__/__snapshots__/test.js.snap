// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity components renders CancelButton correctly 1`] = `
<Component>
  <TouchableOpacity
    activeOpacity={0.8}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e63b2e",
          "elevation": 6,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
        },
        Object {
          "aspectRatio": 1,
          "borderRadius": 53.35999999999999,
          "height": 106.71999999999998,
          "marginRight": 20.009999999999998,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": "35%",
          "tintColor": "#fff",
          "width": "35%",
        }
      }
    />
  </TouchableOpacity>
</Component>
`;

exports[`Activity components renders CompleteButton correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  disabled={false}
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#16db65",
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "aspectRatio": 1,
        "borderRadius": 80.03999999999999,
        "height": 160.07999999999998,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": "50%",
        "tintColor": "#fff",
        "width": "50%",
      }
    }
  />
</TouchableOpacity>
`;

exports[`Activity components renders PauseButton correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#dba159",
        "elevation": 6,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "aspectRatio": 1,
        "borderRadius": 53.35999999999999,
        "height": 106.71999999999998,
        "marginRight": 20.009999999999998,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": "35%",
        "tintColor": "#fff",
        "width": "35%",
      }
    }
  />
</TouchableOpacity>
`;

exports[`Activity components renders ProgressBar correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "backgroundColor": "#c5cbd3",
        "borderRadius": 53.35999999999999,
        "elevation": 6,
        "flexGrow": 1,
        "height": 106.71999999999998,
        "justifyContent": "center",
        "marginRight": 20.009999999999998,
        "overflow": "hidden",
        "position": "relative",
      },
      Object {
        "borderColor": "rgba(0, 0, 0, 0)",
        "borderWidth": 0,
        "elevation": 6,
      },
    ]
  }
>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderColor="red"
    borderRadius={106.71999999999998}
    borderWidth={0}
    color="#474647"
    height={106.71999999999998}
    indeterminate={false}
    progress={0.8333333333333334}
    useNativeDriver={false}
    width={null}
  />
  <Component
    style={
      Array [
        Object {
          "left": "0%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 106.71999999999998,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
  <Component
    style={
      Array [
        Object {
          "left": "50%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 106.71999999999998,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
  <Component
    style={
      Array [
        Object {
          "left": "75%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 106.71999999999998,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
</AnimatedComponent>
`;

exports[`Activity components renders ProgressClock correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 166.07999999999998,
        "bottom": 40.019999999999996,
        "elevation": 6,
        "height": 332.15999999999997,
        "justifyContent": "center",
        "left": 40.019999999999996,
        "position": "absolute",
        "width": 332.15999999999997,
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 1)",
        "elevation": 6,
      },
    ]
  }
>
  <withAnimation(ProgressCircle)
    animated={true}
    borderWidth={0}
    color="#474647"
    indeterminate={false}
    progress={0.8333333333333334}
    size={320.15999999999997}
    thickness={160.07999999999998}
    unfilledColor="#c5cbd3"
  />
</AnimatedComponent>
`;

exports[`Activity components renders Timer correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  -
  1:40
</Text>
`;
